### PR TITLE
Small fixes & improvements to .nomsconfig

### PR DESCRIPTION
**Pull request by ehalpern**
- Fix handling of absolute paths ldb urls
 - Enable for all samples
 - Add -v|--verbose flag to commands to debug expansion
 - Simplify sample README

### CODE REVIEW

**Comment by ehalpern** [comment]
This patch:
  - fixes the problem fully qualified paths
  - enables nomsconfig support for all samples
  - adds a -v|--verbose flag that can be used from any command to see how names are resolved

PTAL @willhite.

File: cmd/noms/noms_serve.go [hunk 1]
```diff
@@ -27,20 +28,25 @@ var nomsServe = &util.Command{
...
 	spec.RegisterDatabaseFlags(serveFlagSet)
+	spec.RegisterVerboseFlags(serveFlagSet)
 	return serveFlagSet
 }
 
 func runServe(args []string) int {
-	spec, err := spec.NewResolver()
-	d.CheckErrorNoUsage(err)
-	cs, err := spec.GetChunkStore(args[0])
+	fmt.Println("Running noms serve")
```
Comment (willhite) [inline cmd/noms/noms_serve.go]: I don't think there's any need for this Println statement.

File: cmd/noms/noms_sync.go [hunk 2]
```diff
@@ -37,22 +37,22 @@ func setupSyncFlags() *flag.FlagSet {
...
 	return syncFlagSet
 }
 
 func runSync(args []string) int {
-	spec, err := spec.NewResolver()
-	d.CheckErrorNoUsage(err)
-	sourceStore, sourceObj, err := spec.GetPath(args[0])
+	resolver := spec.NewResolver()
+	sourceStore, sourceObj, err := resolver.GetPath(args[0])
 	d.CheckError(err)
```
Comment (willhite) [inline cmd/noms/noms_sync.go]: Often times, I see d.CheckError after a resolver failure, and then I see d.CheckErrorNoUsage() after others. Seems like we should do one or the other most of the time.

File: go/spec/config.go [hunk 3]
```diff
@@ -7,26 +7,23 @@ import (
...
 	File    string
-	Default DefaultConfig
 	Db      map[string]DbConfig
 }
 
-type DefaultConfig struct {
-	Url string
-}
-
 type DbConfig struct {
```
Comment (willhite) [inline go/spec/config.go]: Is this type still necessary/desired since it just has one field?

File: go/spec/config.go [hunk 4]
```diff
@@ -100,7 +97,10 @@ func absDbSpec(configHome string, url string) string {
 	if dbSpec.Protocol != "ldb" {
 		return url
 	}
-	path := filepath.Join(configHome, dbSpec.Path)
+	path := dbSpec.Path
+	if !strings.HasPrefix(path, "/") {
+		path = filepath.Join(configHome, path)
+	}
 	if err != nil {
```
Comment (willhite) [inline go/spec/config.go]: What err is getting checked here?

File: samples/cli/nomsconfig/.nomsconfig [hunk 5]
```diff
@@ -1,11 +1,16 @@
 # Default database URL to be used whenever a database is not explictly provided
-[default]
-url = "ldb:.noms/tour"
+[db.default]
+url = "ldb:.noms/tour"  # This path is relative the location of .nomsconfig
```
Comment (willhite) [inline samples/cli/nomsconfig/.nomsconfig]: is relative => is relative to

File: go/spec/verbose.go [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package spec
+
+import (
+	flag "github.com/juju/gnuflag"
+)
+
+var (
```
Comment (willhite) [inline go/spec/verbose.go]: I believe I said this earlier but I wonder if this should move to go/utils rather then live in spec. It's pretty general.

File: samples/cli/nomsconfig/README.md [hunk 7]
```diff
@@ -1,54 +1,60 @@
...
 ```
 
-The *[default]* definition will be used implicitly whenever a database is
-required but not specified. 
+The *[db.default]* section:
 
-The *origin* and *tour* aliases can be used in place of a dataspec.  
+ - Defines a default database
+ - It will be used implicitly whenever a database url is ommitted in a command
+ - It will be used implicitly whenever a database url is ommitted in a command
```
Comment (willhite) [inline samples/cli/nomsconfig/README.md]: Remove one of these duplicate lines.

File: samples/cli/nomsconfig/README.md [hunk 8]
```diff
@@ -1,54 +1,60 @@
...
 
-noms log sf-films                    # noms log ldb:.noms/tour::sf-films
-noms log origin::sf-film-locations   # noms log http://demo.noms.io/cli-tour::sf-film-locations
+noms log sf-films                    # -> noms log ldb:.noms/tour::sf-films
+noms log origin::sf-film-locations   # -> noms log http://demo.noms.io/cli-tour::sf-film-locations
 
-noms show '#1a2aj8svslsu7g8hplsva6oq6iq3ib6c'         # noms show ldb:.noms/tour::'...'
-noms show origin::'#1a2aj8svslsu7g8hplsva6oq6iq3ib6c' # noms show http://demo.noms.io/cli-tour::'...'
+noms show '#1a2aj8svslsu7g8hplsva6oq6iq3ib6c'         # -> noms show ldb:.noms/tour::'...'
+noms show origin::'#1a2aj8svslsu7g8hplsva6oq6iq3ib6c' # -> noms show http://demo.noms.io/cli-tour::'...'
```
Comment (willhite) [inline samples/cli/nomsconfig/README.md]: I'm not sure what the '...' is in these comments. I not certain whether it's a notation your using to avoid retyping the hash or if it's legitimate syntax that I can use on the command line.

File: samples/go/csv/csv-export/perf_test.go [hunk 9]
```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
```
Comment (willhite) [inline samples/go/csv/csv-export/perf_test.go]: Did this file get added by mistake?

File: cmd/noms/noms_commit.go [hunk 10]
```diff
@@ -32,11 +32,13 @@ func setupCommitFlags() *flag.FlagSet {
 	commitFlagSet := flag.NewFlagSet("commit", flag.ExitOnError)
 	commitFlagSet.BoolVar(&allowDupe, "allow-dupe", false, "creates a new commit, even if it would be identical (modulo metadata and parents) to the existing HEAD.")
 	spec.RegisterCommitMetaFlags(commitFlagSet)
+	spec.RegisterVerboseFlags(commitFlagSet)
 	return commitFlagSet
 }
 
 func runCommit(args []string) int {
-	ds, err := spec.GetDataset(args[len(args)-1])
+	resolver := spec.NewResolver()
```
Comment (ehalpern) [inline cmd/noms/noms_commit.go]: Let's walk thru this in person. Hiding resolution in the existing methods is problematic because there are many call points in the code where a config file should not interfere. 

Making the resolver stateless also raises some issues. I think I've got it down to a simple one-liner now, but would love to talk over all the issues with you do determine if there's a better solution.

File: cmd/noms/noms_sync.go (hunk 2 above)
Comment (ehalpern) [inline cmd/noms/noms_sync.go]: The error checking after creating the resolver is removed.

**Comment by ehalpern** [comment]
Jenkins: test this

**Comment by ehalpern** [comment]
jenkins: build this

File: go/util/verbose/verbose.go [hunk 11]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
```
Comment (aboodman) [inline go/util/verbose/verbose.go]: Driveby: I'm glad to see this in `util` :).
